controller/update.go

```diff
@@ -348,30 +337,34 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
 
 					atomic.AddInt64(&addInProgress, -1)
 					done <- struct{}{}
-				}()
+				}(ctx)
```

Comment:
Looks like I was not clear enough. `newReq` is the one we need to pass into the goroutine scope as well.